### ISSUE TITLE
[BUG] listeners do not update if not present in the widget tree

### ISSUE THREAD

**Issue by flmng0** [elkowar/eww#669]
### Checklist before submitting an issue

- [X] I have searched through the existing [closed and open issues](https://github.com/elkowar/eww/issues?q=is%3Aissue) for eww and made sure this is not a duplicate
- [X] I have specifically verified that this bug is not a common [user error](https://github.com/elkowar/eww/issues?q=is%3Aissue+label%3Ano-actual-bug+is%3Aclosed)
- [X] I am providing as much relevant information as I am able to in this bug report (Minimal config to reproduce the issue for example, if applicable)

### Description of the bug

When using a listener, the script does not get listened to, unless it is part of the template tree. 

In the minimal reproduced config, the workspace icons do not update if the line `(revealer :reveal false :transition 'none' bspwm_focused)` is not present.

That line was just a hacky way to force eww to start the listener.

### Reproducing the issue

With the following config:
### eww.yuck
```yuck
(defwindow bar
  :monitor 0
  :geometry (geometry
              :x "0%"
              :y "0%"
              :width "32px"
              :height "100%"
              :anchor "center left")
  ; :stacking "fg"
  ; :wm-ignore false
  :windowtype "dock"
  :reserve (struts :distance "40px" :side "left")
  (bspwm_workspaces))

(defvar bspwm_focused_icon '')
(defvar bspwm_unfocused_icon '')

(deflisten bspwm_desktops
  :initial "[]"
  `scripts/workspaces.fish desktops`)

(deflisten bspwm_focused
  `scripts/workspaces.fish focused`)

(defwidget bspwm_workspaces []
  (box
    :orientation "vertical"
    ; Hidden but necessary. If this is not present, the listener doesn't update.
    (revealer :reveal false :transition 'none' bspwm_focused)
    (for workspace in bspwm_desktops
      (workspace_label 
        :workspace workspace 
        :current bspwm_focused))))
    
(defwidget workspace_label [workspace current]
  (label 
    :class "workspace-label"
    :text {workspace == current ? bspwm_focused_icon : bspwm_unfocused_icon}))
```

### scripts/workspaces.fish
```fish
#!/usr/bin/env fish

function _json_list
    set -l quoted (printf '"%s"\n' $argv | string join ',')

    echo "[$quoted]"
end

function subscribe_focused
    echo (bspc query -D --desktop)

    bspc subscribe desktop_focus | while read -l _ _ desktop
        echo $desktop
    end
end

function _get_desktops
    set -l initial (bspc query -D --monitor default)
    echo (_json_list $initial)
end

function subscribe_desktops
    echo (_get_desktops)

    bspc subscribe desktop | while read -l event
        set -l type (string split ' ' $event)[1]
        if test $type != "desktop_focus"
            echo (_get_desktops)
        end
    end
end

switch $argv[1]
    case focused
        subscribe_focused
    case desktops
        subscribe_desktops
    case '*'
        exit 1
end
```

### Expected behaviour

The icon for a desktop should change to the `bspwm_focused_icon` on change.

### Additional context

Expected behaviour
![image](https://user-images.githubusercontent.com/44032837/213639761-36eccfd7-ef0b-460d-8842-68896ff2049e.png)

Actual behaviour (bspwm_focused variable is not updating):
![image](https://user-images.githubusercontent.com/44032837/213639932-4febef20-9443-48e2-b170-6982e1e3928d.png)
The above image happens when I remove the revealer widget.

**Comment by elkowar** [comment]
Surprisingly, this is actually a feature! The idea is that running variables when the results are never shown should be considered useless overhead, so we just _don't_ run them unless they're used. I'm not 100% clear on your example here, but in your case I'd recommend looking if you can just combine two scripts into one with both of the info, to avoid having this issue. We _may_ want to consider adding a flag to always run variables regardless of if they're used, although I'd argue that that _probably_ hints at a script-structure that is questionable...

**Comment by elkowar** [comment]
If there's any usecase that we genuinely can't fill with this, feel free to re-open or maybe turn this into a feature request, but otherwise I'll close this for now!

**Comment by leg7** [comment]
Imo this is very unexpected behaviour because it breaks the programmers intention. In his example the variable is crucial to display the workspaces but it's never run. The variable is clearly used to display something but it's not run.

I had a very odd issue with this where the listener would not run if I only opened the bar on my main monitor. I had to open it on both monitors for it to run.

If you want I have a simpler example that shows the issue.
